### PR TITLE
new 1.8 version

### PR DESCRIPTION
**Pull request by M-Waszkiewicz-Anaconda**
forex python 1.8 for snowflake
- [upstream](https://github.com/MicroPyramid/forex-python)
- execute only test.py as bitcoin_test.py failing [issue](https://github.com/MicroPyramid/forex-python/issues/147), cannot do otherwise skipping only test_bitcooin.py for some reason

! Tried 1.6 version which according to readme should work: https://github.com/AnacondaRecipes/forex-python-feedstock/pull/2
however it does not, **agreed with snowflake that if earlier version does not work we would deliver the latest even though test are failing**

### CODE REVIEW

**Comment by M-Waszkiewicz-Anaconda** [comment]
looks like the problem is with the underlying services providing forex-data, I have asked snowflake if we should publish even though it does not work at the moment

File: recipe/meta.yaml
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
  url: https://github.com/MicroPyramid/{{ name }}/archive/v{{ version }}.tar.gz
```

**Comment by skupr-anaconda** [comment]
Not a funny warning https://github.com/MicroPyramid/forex-python#forex-python:
`Note: Install latest forex-python==1.6 to avoid RatesNotAvailableError`

See this diff https://github.com/MicroPyramid/forex-python/compare/1.6...v1.8

**Comment by M-Waszkiewicz-Anaconda** [comment]
> Not a funny warning https://github.com/MicroPyramid/forex-python#forex-python: `Note: Install latest forex-python==1.6 to avoid RatesNotAvailableError`
> 
> See this diff [MicroPyramid/forex-python@1.6...v1.8](https://github.com/MicroPyramid/forex-python/compare/1.6...v1.8)

yeah, that is really bad if the latest version does not work, I have asked snowflake what should we do

File: recipe/meta.yaml
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: By using this source the version reported by pip will be 1.6, see https://github.com/MicroPyramid/forex-python/blob/v1.8/setup.py#L5. WHich tells me that the source in GH potentially doesn't match what's on PyPI.

Comment (boldorider4) [reply recipe/meta.yaml]: I think it would be appropriate to patch that `setup.py` file.

File: recipe/meta.yaml
Comment (boldorider4) [inline recipe/meta.yaml]: `requests` is already in run

File: recipe/meta.yaml
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml]: I would actually diff the source files to see if there is any other difference.

File: recipe/meta.yaml
Comment (boldorider4) [inline recipe/meta.yaml]: if tests are failing for specific reason, I think it's good to document them here and still run `pytest tests`.
Also, tests have been added as `source_files`.

File: recipe/meta.yaml
Comment (boldorider4) [reply recipe/meta.yaml]: The tags `1.6` and `v1.8` are few commits apart in the history so they have to differ. Weird that `v1.7` and `v1.8` point to the same commit. In any case, whatever the upstream developer considers 1.8, we can only assume is what we have to pack.

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml]: I don't trust upstream in this case. If GH and PyPI differ, we need to look at the diff and decide what to do.

Comment (boldorider4) [reply recipe/meta.yaml]: I thought PyPi just redistributes whatever comes from GH. Maybe still worth to check though.

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml]: PyPI has no notion of what GH is and nothin stops a maintainer from releasing any code from anywhere.

Comment (boldorider4) [reply recipe/meta.yaml]: PyPi links PR, issues, forks, etc. from the GH project so it should have notion of it, but ok, I get that the releases may still not come from GH.

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml]: >PyPi links PR, issues, forks, etc

People add links via package metadata. PyPI only adds icons to links to links provided in package metadata.

File: recipe/meta.yaml
Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml]: I will put it on hold until snowflake tell us what they want

File: recipe/meta.yaml
Comment (boldorider4) [reply recipe/meta.yaml]: So your concern is that someone may have uploaded different source code than GH, albeit the links that come from the package metadata point to the GH project. It sounds like a really weird scenario, but I do agree it's worth to check for differences.

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml]: The link is just metadata. It's not a guarantee that the code comes from GH. It's basically https://github.com/MicroPyramid/forex-python/blob/master/setup.py#L27.

Comment (cbouss) [reply recipe/meta.yaml]: It doesn't happen often, but it does happen.

**Review by skupr-anaconda — APPROVED** [review]
Looks good but small fix

File: recipe/meta.yaml
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml]: I continue to think that we should take the source code from PyPI.

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Pull request merged**
